hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/container/TestContainerReportHandlingWithHA.java

```diff
@@ -113,23 +110,7 @@ void testDeletingOrDeletedContainerTransitionsToClosedWhenNonEmptyReplicaIsRepor
         // restart a DN and wait for the container to get CLOSED in all SCMs
         HddsDatanodeService dn = cluster.getHddsDatanode(keyLocation.getPipeline().getFirstNode());
         cluster.restartHddsDatanode(dn.getDatanodeDetails(), false);
-        ContainerManager[] array = new ContainerManager[numSCM];
-        for (int i = 0; i < numSCM; i++) {
-          array[i] = cluster.getStorageContainerManager(i).getContainerManager();
-        }
-        GenericTestUtils.waitFor(() -> {
-          try {
-            for (ContainerManager manager : array) {
-              if (manager.getContainer(containerID).getState() != HddsProtos.LifeCycleState.CLOSED) {
-                return false;
-              }
-            }
-            return true;
-          } catch (ContainerNotFoundException e) {
-            fail(e);
-          }
-          return false;
-        }, 2000, 20000);
+        waitForContainerStateInSCM(cluster.getScmLeader(), containerID, HddsProtos.LifeCycleState.CLOSED);
```

Comment:
The point of this test is that the container should eventually move to `CLOSED` in all three SCMs, so we should check the other SCMs as well.